esp-wifi/src/compat/common.rs

```diff
@@ -201,15 +182,30 @@ pub fn thread_sem_get() -> *mut c_void {
 
 pub fn create_recursive_mutex() -> *mut c_void {
     critical_section::with(|_| unsafe {
-        let ptr = &mut MUTEXES[MUTEX_IDX_CURRENT] as *mut Mutex;
-        (*ptr).recursive = true;
-        MUTEX_IDX_CURRENT += 1;
+        let mutex = Mutex {
+            locking_pid: 0xffff_ffff,
+            count: 0,
+            recursive: true,
+        };
+
+        let ptr = unsafe { malloc(size_of_val(&mutex) as u32) as *mut Mutex };
+        unsafe {
+            ptr.write(mutex);
+        }
         memory_fence();
+
         trace!("recursive_mutex_create called {:?}", ptr);
         ptr as *mut c_void
     })
 }
 
+pub fn mutex_delete(mutex: *mut c_void) {
+    let ptr = mutex as *mut Mutex;
+    critical_section::with(|_| unsafe {
```

Comment:
Why does this have to be done in a cs?

Reply:
true - we don't need that anymore - IIRC we needed it before when we directly used linked-list-allocator